framework/core/src/Api/Controller/ListPostsController.php

```diff
@@ -145,7 +146,9 @@ protected function extractOffset(ServerRequestInterface $request)
                 );
             }
 
-            $offset = $this->posts->getIndexForNumber($filter['discussion'], $near, $actor);
+            $discussionId = Discussion::findOrFail((int) $filter['discussion'])->id;
+
+            $offset = $this->posts->getIndexForNumber($discussionId, $near, $actor);
```

Comment:
I'm not sure if we want to error here: this would disclose whether a discussion exists or not, regardless of whether the actor has access.

Reply:
You're right, instead, we should just return a 0 offset.